create-mango/source/create-project.ts

```diff
@@ -173,6 +182,35 @@ async function createProject(options: Options) {
 				return execaInDirectory('npx', ['prettier', '--write', '.'])
 			},
 		},
+		{
+			title: 'Initialize Git repository',
+			enabled: async () => {
+				try {
+					await execaInDirectory('git', ['rev-parse', '--is-inside-work-tree'])
+					return false
+				} catch (error) {
+					return error.message.includes('not a git repository')
+				}
+			},
+			task: async () => {
+				// The rest of your Git initialization logic
+				await execaInDirectory('git', ['init'], {
+					cwd: projectDirectoryPath,
+				})
+
+				await execaInDirectory('git', ['add', '.'], {
+					cwd: projectDirectoryPath,
+				})
+
+				await execaInDirectory(
+					'git',
+					['commit', '-m', 'Initial commit from Mangobase'],
+					{
+						cwd: projectDirectoryPath,
+					}
+				)
+			},
+		},
```

Comment:
This creates the git repo regardless. I think the `Promise<>` API for this doesn't work properly.

Can you move this function out of here then do something like

```javascript
async function shouldInitGit() {
   ...
}

const initGit = await shouldInitGit()

...
{
  title: 'Initialize Git rep...',
  enabled: initGit,
  ...
}

Reply:
I tried submitting a PR against your branch, but it's not up to date with master so has many changes appearing the diff